components/eventing-controller/pkg/handlers/jetstream.go

```diff
@@ -45,12 +47,49 @@ type JetStreamBackend interface {
 	GetJetStreamSubjects(subjects []string) []string
 }
 
+// SubscriptionSubjectIdentifier is used to uniquely identify a Subscription subject.
+// It should be used only with JetStream backend.
+type SubscriptionSubjectIdentifier struct {
+	consumerName, namespacedName string
+}
+
+// NewSubscriptionSubjectIdentifier returns a new SubscriptionSubjectIdentifier instance.
+func NewSubscriptionSubjectIdentifier(subscription *eventingv1alpha1.Subscription, subject string) SubscriptionSubjectIdentifier {
+	cn := computeConsumerName(subscription, subject) // compute the consumer name once
+	nn := computeNamespacedName(subscription)        // compute the namespaced name once
+	return SubscriptionSubjectIdentifier{consumerName: cn, namespacedName: nn}
+}
+
+// ConsumerName returns the JetStream consumer name.
+func (s SubscriptionSubjectIdentifier) ConsumerName() string {
+	return s.consumerName
+}
+
+// NamespacedName returns the Kubernetes namespaced name.
+func (s SubscriptionSubjectIdentifier) NamespacedName() string {
+	return s.namespacedName
+}
+
+// computeConsumerName returns JetStream consumer name of the given subscription and subject.
+// It uses the crypto/md5 lib to return a string of 32 characters as recommended by the JetStream
+// documentation https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming.
+func computeConsumerName(subscription *eventingv1alpha1.Subscription, subject string) string {
+	cn := subscription.Namespace + separator + subscription.Name + separator + subject
```

Comment:
(nit) is it better to use `fmt.sprintf()`?

Reply:
As discussed offline, we'll keep it like that because simple string concatenation is more performant that `fmt.sprintf()`.